Utilities/build-script-helper.py

```diff
@@ -65,8 +91,12 @@ def get_swiftpm_options(args):
       '-Xlinker', '-rpath', '-Xlinker', '$ORIGIN/../lib/swift/linux',
     ]
 
+  build_target = get_build_target(swift_exec, args)
   if args.cross_compile_host:
-    swiftpm_args += ['--destination', args.cross_compile_config]
+    if build_target == 'x86_64-apple-macosx' and args.cross_compile_host == "macosx-arm64":
+        swiftpm_args += ["--arch", "x86_64", "--arch", "arm64"]
```

Comment:
Would you bring the two Android lines deleted below up here and move the config line deleted two lines above after printing the cross-compilation host? Otherwise, this is going to break [cross-compiling sourcekit-lsp for Android](https://github.com/termux/termux-packages/blob/master/packages/swift/build.sh#L162).